tests/profiling/collector/pprof_utils.py

```diff
@@ -28,19 +30,33 @@ def _protobuf_version() -> Tuple[int, int, int]:
     return parse_version(google.protobuf.__version__)
 
 
-# Load the appropriate pprof_pb2 module
+# Load the appropriate pprof_pb2 module based on protobuf version
 _pb_version = _protobuf_version()
-for v in [(4, 21), (3, 19), (3, 12)]:
-    if _pb_version >= v:
-        import sys
-
-        pprof_module = "tests.profiling.collector.pprof_%s%s_pb2" % v
-        __import__(pprof_module)
-        pprof_pb2 = sys.modules[pprof_module]
-        break
+if _pb_version >= (4, 21):
+    from tests.profiling.collector import pprof_421_pb2 as pprof_pb2
+elif _pb_version >= (3, 19):
+    from tests.profiling.collector import pprof_319_pb2 as pprof_pb2  # type: ignore[no-redef]
+elif _pb_version >= (3, 12):
+    from tests.profiling.collector import pprof_312_pb2 as pprof_pb2  # type: ignore[no-redef]
 else:
     from tests.profiling.collector import pprof_3_pb2 as pprof_pb2  # type: ignore[no-redef]
 
+if TYPE_CHECKING:
+    # For type checking, use the actual protobuf types
+    # Protobuf-generated modules use dynamic attributes, so we need type: ignore
+    PprofProfile: TypeAlias = pprof_pb2.Profile  # type: ignore[name-defined,attr-defined]
+    PprofSample: TypeAlias = pprof_pb2.Sample  # type: ignore[name-defined,attr-defined]
+    PprofLabel: TypeAlias = pprof_pb2.Label  # type: ignore[name-defined,attr-defined]
+    PprofLocation: TypeAlias = pprof_pb2.Location  # type: ignore[name-defined,attr-defined]
+    PprofFunction: TypeAlias = pprof_pb2.Function  # type: ignore[name-defined,attr-defined]
+else:
+    # At runtime, use Any for type annotations since types are determined dynamically
+    PprofProfile: TypeAlias = Any
+    PprofSample: TypeAlias = Any
+    PprofLabel: TypeAlias = Any
+    PprofLocation: TypeAlias = Any
+    PprofFunction: TypeAlias = Any
```

Comment:
I'm not a python types expert.  This is what the AI suggested: does it make sense?

Reply:
done!